### ISSUE TITLE
Project restructuring sandbox

### ISSUE THREAD

**Issue by kaisalmen** [TypeFox/monaco-languageclient#331]
This issue covers "Required next work:" from #315 
* [x] Some tools/libraries are now quite old (typescript for instance)
* [x] The build has unexpected problems (https://github.com/TypeFox/monaco-languageclient/issues/321)
* [x] Do we really need lerna? The only thing being released is the client. Would it make sense to switch to npm workspaces (this options wasn't available in 2018, of course)?
* [x] Verify build on every commit: with GitHub Actions. (https://github.com/TypeFox/monaco-languageclient/actions)
* [x] Switch dependency of `monaco-editor-core` to`monaco-editor` and make dependency obvious (see #337)

I will start a new branch and test things ([`sandbox-331`](https://github.com/TypeFox/monaco-languageclient/tree/sandbox-331)).

**Comment by kaisalmen** [comment]
@CGNonofr We have GitHub Actions! 🎉

**Comment by CGNonofr** [comment]
To not poluate the root folder with too many subfolders, I would have put everything inside an `example` folder

```
packages/client // npm module
packages/example/node // server part
packages/example/client // browser client
packages/example/browser // complete in browser
```
(Even If I'm not sure about the relevance of the browser example)

I see no problem from mergeing to whatever bundler you want, but is there is reason? is there an issue with webpack?

**Comment by kaisalmen** [comment]
> (Even If I'm not sure about the relevance of the browser example)

Let's not waste what is already there, right?

Currently webpack build is broken on my branch after upgrading versions. vite is just fast and easier to handle, but that's personal preference, of course. We don't need production builds for the examples and here plain node (server) and vite for client/browser related stuff is faster and easier to configure.

**Comment by kaisalmen** [comment]
@CGNonofr Making good progress now. client itself and examples/node and examples/client have been stripped of unneeded dependencies. Using monaco-editor now in the client example!

**Comment by CGNonofr** [comment]
Will you do a single PR including all your changes?

What do you expect the work to be done?

**Comment by kaisalmen** [comment]
> Will you do a single PR including all your changes?

I fear yes, because I don't know how to separate the things properly.

> What do you expect the work to be done?

Maybe today you can have a first look (draft PR).

**Comment by kaisalmen** [comment]
Ok, all examples work: https://github.com/TypeFox/monaco-languageclient/tree/sandbox-331
I need to update the README and will then open a draft PR.
If you want to test it yourself (gitpod configuration is already adjusted). Best with separate checkout:
``` shell
# this needs to be done 
npm i
# cleans and compiles everything
npm run build 
# start the node LS server embedded or external
npm run start-example-node
npm run start-example-node:ext
# run vite in debug mode for both client based examples
npm run dev
```
Actually for vite to work the client and browser example don't need to be compiled.
I have created a new project `monaco-editor-workers` that is used to configure workers (classic or module workers) in the `MonacoEnvironment` , so no-one needs to think about this stuff. It is applied in the client and browser example.

**Comment by kaisalmen** [comment]
Resolved with #340
